activemodel/lib/active_model/dirty.rb

```diff
@@ -139,6 +139,13 @@ def initialize_dup(other) # :nodoc:
       @mutations_from_database = nil
     end
 
+    # Exclude +mutations_from_database+ attribute from json as it lead to
+    # recursive call to as_json method
+    def as_json(options = {})
+      options[:except] = "mutations_from_database"
```

Comment:
@anilmaurya Do any of the other options cause recursion issues?

Reply:
No, as far as I know.